### PR TITLE
ci: move audit check to cron

### PR DESCRIPTION
**Pull request by xxchan**
no need to run it per pr, which usually blocks all pr unnecessarily

I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added test labels as necessary. See [details](https://github.com/risingwavelabs/risingwave/blob/main/docs/developer-guide.md#ci-labels-guide).
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [ ] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

</details>

### CODE REVIEW

File: ci/workflows/main-cron.yml [hunk 1]
```diff
@@ -326,7 +326,7 @@ steps:
     retry: *auto-retry
 
   - label: "misc check"
-    command: "ci/scripts/misc-check.sh"
+    command: "ci/scripts/misc-check-cron.sh"
```
Comment (xxchan) [inline ci/workflows/main-cron.yml:329]: `misc-check.sh` contains `buf format & lint`, no need to run it in cron..

File: ci/scripts/misc-check-cron.sh [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+set -euo pipefail
+
+source ci/scripts/common.sh
+
+# RUSTSEC-2023-0052: https://github.com/risingwavelabs/risingwave/issues/11842
+# RUSTSEC-2023-0071 https://github.com/risingwavelabs/risingwave/issues/13703
+echo "--- Run audit check"
+cargo audit \
+  --ignore RUSTSEC-2023-0052 \
+  --ignore RUSTSEC-2023-0071
```
Comment (kwannoel) [inline ci/scripts/misc-check-cron.sh:13]: Could we use a config file for it?

https://github.com/risingwavelabs/risingwave/pull/13716/files

**Review by kwannoel — APPROVED** [review]
LGTM, thanks

**Review by fuyufjh — APPROVED** [review]

**Comment by xxchan** [comment]
wait, we can use GHA instead

**Review by BugenZhao — APPROVED** [review]

**Pull request merged**